pkg/runner/client.go

```diff
@@ -165,11 +165,13 @@ func CreateClient(r prometheus.Registerer, cfg *Config) (*pgclient.Client, error
 		cfg.APICfg.MultiTenancy = multiTenancy
 	}
 
-	if cfg.DatasetConfig != "" {
+	if (cfg.DatasetCfg != dataset.Config{}) {
+		err = cfg.DatasetCfg.Apply(conn)
+	} else if cfg.DatasetConfig != "" {
```

Comment:
I'm keeping both `startup.dataset.config` (the old string one) and `startup.dataset` (new one that's a yaml mapping).  We can add a deprecation notice and remove `startup.dataset.config` later.

I think the suffix `config` here is redundant since these are all config options. This gives us an opportunity to fix that and don't  cause a breaking change.

Other options are:

- Change `startup.dataset.config` to accept the yaml mapping directly. 
- Don't add deprecation notice and just remove the old config option `startup.dataset.config`

Reply:
This was actually supposed to be startup.dataset_config. another words these options are "dataset configs" but I don't love the name and am up for something else (although I think dataset by itself reads weird)

Reply:
The only problem I see with keeping both options is that you can use them both at the same time and create confusion for the user on which one is actually being used to configure. We might need to validate that case and notify or maybe even force using just one option at a time.

Reply:
I’ll add warning if both are being used, saying which one has precedence and a deprecation warning for users that have the old one